stdlib/http/src/main/ballerina/src/http/client_endpoint.bal

```diff
@@ -29,22 +29,26 @@ import ballerina/time;
 # + config - The configurations associated with the client
 # + httpClient - Chain of different HTTP clients which provides the capability for initiating contact with a remote
 #                HTTP service in resilient manner
+# + cookieStore - Stores the cookies of the client
 public type Client client object {
 
     public string url;
     public ClientConfiguration config = {};
     public HttpClient httpClient;
+    public CookieStore cookieStore;
 
-    # Gets invoked to initialize the client. During initialization, configurations provided through the `config`
-    # record is used to determine which type of additional behaviours are added to the endpoint (e.g: caching,
+    # Gets invoked to initialize the client. During initialization, the configurations provided through the `config`
+    # record is used to determine which type of additional behaviours are added to the endpoint (e.g., caching,
     # security, circuit breaking).
     #
     # + url - URL of the target service
     # + config - The configurations to be used when initializing the client
+    # + cookieStore - Stores the cookies of the client
     public function __init(string url, public ClientConfiguration? config = ()) {
         self.config = config ?: {};
         self.url = url;
-        var result = initialize(url, self.config);
+        self.cookieStore = new;
```

Comment:
Do we need to initialise a cookie store with every client creation? Is it this optional?

Reply:
Every client gets a cookie store even though he is a cookie client or not.So, we have to initialize the cookie store with every client.If not,  the client can't access to his cookie store.

Reply:
But IMO, this is an unnecessary memory usage. @arukshani  @shafreenAnfar WDYT?

Reply:
@Tharuja Can you initialize the cookie store only when it is enabled in the config? That way we will only create a cookie store when it is needed.

Reply:
+1 , Sure will do.